beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -41,6 +41,8 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
 	ctx, span := trace.StartSpan(ctx, "ProposerServer.GetBeaconBlock")
 	defer span.End()
 	span.AddAttributes(trace.Int64Attribute("slot", int64(req.Slot)))
+	vs.ForkChoiceLocker.Lock()
+	defer vs.ForkChoiceLocker.Unlock()
```

Comment:
I feel a bit strongly about reducing the surface area for this lock. We should be able to release it after `HeadState`. Is there a reason why we are not releasing it earlier

Reply:
To answer my question, simplicity and clarity is one reasons not to do it

Reply:
Also `circuitBreakBuilder` uses head fetcher as well. I think it's best to keep it as it is